boards/nordic/nrf9280pdk/nrf9280pdk_nrf9280-memory_map.dtsi

```diff
@@ -0,0 +1,271 @@
+/*
+ * Copyright (c) 2024 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+#include <zephyr/dt-bindings/memory-attr/memory-attr.h>
+
+/ {
+	reserved-memory {
+		/* The first 64kb are reserved for SecDom.
+		 * The next 4kb are reserved for IPC between SecDom and Cellcore.
+		 */
+
+		cpurad_ram0x_region: memory@2f011000 {
+			compatible = "nordic,owned-memory";
+			reg = <0x2f011000 DT_SIZE_K(4)>;
+			status = "disabled";
+			perm-read;
+			perm-write;
+			perm-secure;
+			#address-cells = <1>;
+			#size-cells = <1>;
+			ranges = <0x0 0x2f011000 0x1000>;
+
+			cpusec_cpurad_ipc_shm: memory@0 {
+				reg = <0x0 DT_SIZE_K(2)>;
+			};
+
+			cpurad_cpusec_ipc_shm: memory@800 {
+				reg = <0x800 DT_SIZE_K(2)>;
+			};
+		};
+
+		cpuapp_ram0x_region: memory@2f012000 {
+			compatible = "nordic,owned-memory";
+			reg = <0x2f012000 DT_SIZE_K(516)>;
+			status = "disabled";
+			perm-read;
+			perm-write;
+			perm-secure;
+			#address-cells = <1>;
+			#size-cells = <1>;
+			ranges = <0x0 0x2f012000 0x81000>;
+
+			cpusec_cpuapp_ipc_shm: memory@0 {
+				reg = <0x0 DT_SIZE_K(2)>;
+			};
+
+			cpuapp_cpusec_ipc_shm: memory@800 {
+				reg = <0x800 DT_SIZE_K(2)>;
+			};
+
+			cpuapp_data: memory@1000 {
+				reg = <0x1000 DT_SIZE_K(512)>;
+			};
+		};
+
+		cpuapp_cpurad_ram0x_region: memory@2f0cf000 {
+			compatible = "nordic,owned-memory";
+			reg = <0x2f0cf000 DT_SIZE_K(4)>;
+			status = "disabled";
+			perm-read;
+			perm-write;
+			#address-cells = <1>;
+			#size-cells = <1>;
+			ranges = <0x0 0x2f0cf000 0x1000>;
+
+			cpuapp_cpurad_ipc_shm: memory@0 {
+				reg = <0x0 DT_SIZE_K(2)>;
+			};
+
+			cpurad_cpuapp_ipc_shm: memory@800 {
+				reg = <0x800 DT_SIZE_K(2)>;
+			};
+		};
+
+		/* TODO:
+		 * refactor after https://github.com/zephyrproject-rtos/zephyr/pull/72273
+		 * to use the `access` binding to specify shared memory ownership.
+		 */
+		cpuapp_cpucell_ram0x_region: memory@2f0d0000 {
+			reg = <0x2f0d0000 DT_SIZE_K(36)>;
+			status = "disabled";
+			#address-cells = <1>;
+			#size-cells = <1>;
+			ranges = <0x0 0x2f0d0000 0x9000>;
+
+			/* Control region, with ICmsg buffers.
+			 * Size is fixed.
+			 */
+			cpuapp_cpucell_ipc_shm_ctrl: memory@0 {
+				reg = <0x0 0x1000>;
+			};
+			/* TX heap, user defined */
+			cpuapp_cpucell_ipc_shm_heap: memory@1000 {
+				reg = <0x1000 0x4000>;
+			};
+			/* RX heap, user defined */
+			cpucell_cpuapp_ipc_shm_heap: memory@5000 {
+				reg = <0x5000 0x4000>;
+			};
+		};
+
+		/* Shared memory ownership */
+		cpuapp_cpucell_ipc_shm: memory@2 {
+			compatible = "nordic,owned-memory";
+			reg = <0x2f0d0000 DT_SIZE_K(36)>;
+			owner-id = <2>;
+			perm-read;
+			perm-write;
+			status = "disabled";
+		};
+		cpucell_cpuapp_ipc_shm: memory@4 {
```

Comment:
The unit address is wrong; it needs to match the value of `reg`. Be careful though: we can't have two nodes with the same name - `memory@2f0d0000` - so this one must be different somehow.
```suggestion
		cpucell_cpuapp_ipc_shm: memory-alt@2f0d0000 {
```

Reply:
Agreed offline to keep as-is, given that it does not actually generate proper warnings (just messages), unlike having two nodes with the same address.

Reply:
For reference, the messages that currently get printed:
```
unit address and first address in 'reg' (0x2f0d0000) don't match for /reserved-memory/memory@2
unit address and first address in 'reg' (0x2f0d0000) don't match for /reserved-memory/memory@4
```
are edtlib warnings, but they should be harmless. They were meant to replicate the `simple_bus_reg` warning from dtc, but they do so incorrectly, which is why the actual dtc warning doesn't show. See #57059